core/src/Streamly/Internal/Data/Stream/Serial.hs

```diff
@@ -131,19 +115,19 @@ type Serial = SerialT IO
 
 {-# INLINE cons #-}
 cons :: a -> SerialT m a -> SerialT m a
-cons x (SerialT ms) = SerialT $ K.cons x ms
+cons x st = Stream.fromStreamK $ K.cons x (Stream.toStreamK st)
 
 {-# INLINE consM #-}
 {-# SPECIALIZE consM :: IO a -> SerialT IO a -> SerialT IO a #-}
 consM :: Monad m => m a -> SerialT m a -> SerialT m a
-consM m (SerialT ms) = SerialT $ K.consM m ms
+consM m st = Stream.fromStreamK $ K.consM m (Stream.toStreamK st)
 
 -- |
 -- Generate an infinite stream by repeating a pure value.
 --
 {-# INLINE_NORMAL repeat #-}
 repeat :: Monad m => a -> SerialT m a
-repeat = SerialT . D.toStreamK . D.repeat
+repeat = Stream.fromStreamK . D.toStreamK . D.repeat
```

Comment:
Can this be moved to `Stream`? Do we want to remove it and use unfolds instead? But since we have the `use-unfolds` flag, we might want to keep this for benchmark comparison purposes.

Reply:
Should not remove it yet. This internal module may be used some consumers. We will deprecate this module and then remove later.